overview.tt

```diff
@@ -0,0 +1,279 @@
+        </p>
+        <p>
+          Forget <code>Dockerfile</code> and
+          have a peace of mind with declarative approach Nix gives you. Nix
+          automatically calculates the optimal layer layout and only includes
+          software needed.
+          <strong>Nix and Docker is a match made in heaven.</strong>
+        </p>
+      </div>
+    </div>
+  </div>
+
+  <div>
+    [% PROCESS svg path="site-styles/assets/gfx-overview-gears.svg" %]
+    <div>
+      <h2>
+        Is the previous version better? Rollback anytime.
+      </h2>
+      <p>
+        Packages are never overwritten, and old versions are always stored.
+        Rollback to any previous version within milliseconds.
+      </p>
+      <ul>
+        <li>generations</li>
+        <li>garbage collection</li>
+        <li>atomic</li>
+        <li>nix store</li>
+      </ul>
+      <div>
+        <p>
+          Since Nix never overwrites files in the Nix store, but just adds new
+          versions in different paths, the upgrades are atomic. During an
+          upgrade, there is no time window in which some files exists from the
+          old version and some files from the new version - which would be bad
+          because a program might crash if it's restarted during that period.
+        </p>
+        <p>
+          And since nothing is overwritten, the old versions are still there
+          after an upgrade. This means that you can rollback to the old
+          version. Rollbacks are very fast: it doesn't involve lots of files
+          having to be restored from copies.
+        </p>
+      </div>
+    </div>
+  </div>
+
+  <div>
+    <!-- hacking, fun, party -->
+    [% PROCESS svg path="site-styles/assets/gfx-overview-gears.svg" %]
+    <div>
+      <h2>
+        It is important to experiment.
+      </h2>
+      <ul>
+        <li>generations</li>
+        <li>nix store</li>
+      </ul>
+      <p>
+        Use tools that allow you to play with different software as
+        you did when the job wasn't waiting for you the next day.
+      </p>
+      <div>
+        <p>
+          Install new software with no risk of breaking anything down. Nix
+          prevents conflicts not only between multiple development environments,
+          but also with the operating system. Your work environment will stay
+          intact, but allow you to experiment in a safe way.
+        </p>
+        <p>
+          No need to turn your system upside down just because you want to test
+          some experimental and bleeding edge versions of software.
+        </p>
+      </div>
+    </div>
+  </div>
+
+  <div>
+    [% PROCESS svg path="site-styles/assets/gfx-overview-gears.svg" %]
+    <div>
+      <h2>
+        No surprises. Trust in your deployments.
+      </h2>
+      <p>
+        Surprises only belong to birthdays, not deployments. With Nix there
+        are only two possible outcomes. Your system was successfully upgraded
+        or upgrade did not happen.
```

Comment:
This is a pretty strong claim. It's true for `nix-env` but not for NixOS upgrades because they do have state that can cause an upgrade to break. And we are mentioning NixOS in this item...